sweep/sweeper.go

```diff
@@ -949,6 +981,135 @@ func (s *UtxoSweeper) handlePendingSweepsReq(req *pendingSweepsReq) []*PendingIn
+// replacement transaction (RBF) will be created to allow the new fee rate to
+// propagate throughout the network, or a child transaction will be created to
+// cover its parents fees (CPFP).
+//
+// NOTE: This currently doesn't do any fee rate validation to ensure that a bump
+// is actually successful. The responsibility of doing so should be handled by
+// the caller.
+func (s *UtxoSweeper) BumpFee(input input.Input,
+	feePreference FeePreference) (chan Result, error) {
+
+	if input == nil || input.OutPoint() == nil {
+		return nil, errors.New("nil input received")
+	}
+
+	// Ensure the client provided a sane fee preference.
+	if _, err := s.feeRateForPreference(feePreference); err != nil {
+		return nil, err
+	}
+
+	responseChan := make(chan *bumpFeeResp, 1)
+	select {
+	case s.bumpFeeReqs <- &bumpFeeReq{
+		input:         input,
+		feePreference: feePreference,
+		responseChan:  responseChan,
+	}:
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+
+	select {
+	case response := <-responseChan:
+		return response.resultChan, response.err
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+}
+
+// handleBumpFee handles an incoming request to bump the fee rate of a given
+// input. Depending on the UtxoSweeper's knowledge of the input, either a
+// replacement transaction (RBF) will be created to allow the new fee rate to
+// propagate throughout the network, or a child transaction will be created to
+// cover its parents fees (CPFP).
+func (s *UtxoSweeper) handleBumpFeeReq(req *bumpFeeReq,
+	bestHeight int32) (chan Result, error) {
+
+	// If the UtxoSweeper is already trying to sweep this input, then we can
+	// simply just increase its fee rate. This will allow the input to be
+	// batched with others which also have a similar fee rate, creating a
+	// higher fee rate transaction that replaces the original input's
+	// sweeping transaction.
+	if pendingInput, ok := s.pendingInputs[*req.input.OutPoint()]; ok {
+		return s.handleRBFBump(req, pendingInput, bestHeight)
+	}
+
+	// If the input is unknown to the UtxoSweeper and we don't have
+	// information about the output needed to properly sweep it, then we'll
+	// return an error indicating so.
+	if req.input.SignDesc().Output == nil {
+		return nil, lnwallet.ErrNotMine
+	}
+
+	// Otherwise, we'll register the input within the UtxoSweeper, which
+	// will end up creating a child transaction that pays for its parent's
+	// fee rate.
+	return s.handleCPFPBump(req, bestHeight)
+}
+
+// handleRBFBump handles a bump fee request by simply updating the inputs fee
+// preference. Currently, no validation is done on the new fee preference to
+// ensure it will properly create a replacement transaction.
+//
+// TODO(wilmer):
+//   * Validate fee preference to ensure we'll create a valid replacement
+//     transaction to allow the new fee rate to propagate throughout the
+//     network.
+//   * Ensure we don't combine this input with any other unconfirmed inputs that
```

Comment:
Are there any situations that exist today, taking into account our current use cases, wherein improper usage of this new RPC can cause this to happen?

Reply:
It can happen if the sweeper increases the fee rate of an input and a CPFP is made where the input being swept is within the same bucket as the increased fee rate input.